dev/build_release.py

```diff
@@ -339,42 +313,118 @@ def start_container_build(self, name):
+    cmds = [
+      ('gcloud container builds submit --account={account} --project={project} --config"../{name}-gcb.yml"'
+       .format(name=name, account=gcb_service_account, project=gcb_project))
+    ]
+    logfile = '{name}-gcb-build.log'.format(name=name)
+    run_shell_and_log(cmds, logfile, cwd=gradle_root)
+
+  @classmethod
+  def __gcb_trigger_build(cls, name, gradle_root, gcb_service_account, gcb_service_account_json, gcb_project, mirror_base_url):
+    tag = run_quick('cat {name}-gcb-trigger.yml'.format(name=name), echo=False).stdout.strip()
+    cmds = [
+      'git remote add mirror {base_url}/{name}.git'.format(base_url=mirror_base_url, name=name),
+      'git fetch mirror',
+      'git checkout mirror/master',
+      'git merge origin/master',
+      'git push mirror master',
+      'git push mirror {tag}'.format(name=name, tag=tag)
+    ]
+    logfile = '{name}-gcb-triggered-build.log'.format(name=name)
+    run_shell_and_log(cmds, logfile, cwd=gradle_root)
+
+    pubsub_client = pubsub.Client.from_service_account_json(gcb_service_account_json)
+    topic = pubsub_client.topic('cloud-builds') # GCB creates a topic named this automatically.
+    print 'Creating subscription: cloud-builds-{}'.format(name)
+    subscription = topic.subscription('cloud-builds-{name}'.format(name=name))
+    subscription.create()
+    if not subscription.exists():
+      raise LookupError('GCB pubsub subscription creation for subscription id {} failed.'.format(subscription.name))
+
+    def fail_build(name):
+      raise Exception('GCB triggered build for {} timed out'.format(name))
+
+    # Set an egg timer to fail.
+    timer = threading.Timer(1200, fail_build, (name))
+    timer.start()
+
+    # Poll Google Cloud Pubsub for the build status
+    completed = False
+    try:
+      while not completed:
+        pulled = subscription.pull()
+        for ack_id, message in pulled:
+          comp_name = ''
+          if name == 'spinnaker-monitoring':
+            comp_name = 'monitoring-daemon'
+          else:
+            comp_name = name
+          payload = json.loads(message.data)
+          repo_name = payload['source']['repoSource']['repoName']
+          tag_name = payload['source']['repoSource']['tagName']
+          if repo_name == comp_name and tag_name == tag:
+            subscription.acknowledge([ack_id])
+            status = payload['status']
+            print 'Received status: {} for building tag {} of {}'.format(status, tag_name, comp_name)
+            if status in ['SUCCESS', 'FAILURE']:
+              completed = True
+              build_id = payload['id']
+              print 'Retrieving logs for build_id: {}'.format(build_id)
+              get_log_cmd = ('gcloud container builds log --project {project} --account {account} {id}'
+                             .format(project=gcb_project, account=gcb_service_account, id=build_id))
+              build_log = run_quick(get_log_cmd, echo=False).stdout.strip()
```

Comment:
How big is the buffer? These are generally around 100 lines. I didn't notice any issues in my testing.